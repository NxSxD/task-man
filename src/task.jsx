import React from 'react';
import styled from 'styled-components';

export const Task = ({ task, ...restOfProps }) => {
  return (
    <TaskContainer {...restOfProps}>
      <TaskTitle>{task.title}</TaskTitle>
      <TaskDescription>{task.description}</TaskDescription>
    </TaskContainer>
  );
};

const TaskContainer = styled.div`
  margin: 8px;
  background: #fff;
  height: 120px;
  width: 120px;
  padding: 8px;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  &:hover {
    cursor: pointer;
  }
`;

const TaskTitle = styled.span`
  font-size: 18px;
  color: #444;
`;

const TaskDescription = styled.span`
  font-size: 14px;
  color: #666;
`;
