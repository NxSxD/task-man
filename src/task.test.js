import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { Task } from './task';

const task = {
  id: 'react',
  title: 'Learn React',
  description: 'Learn react & redux',
};

describe('Task', () => {
  const onComplete = jest.fn();
  it('should render correctly according to snapshot', () => {
    const element_tree = render(<Task task={task} />);

    expect(element_tree.container).toMatchSnapshot();
  });

  it('should render Title', () => {
    const { getByText } = render(<Task task={task} />);
    expect(getByText(new RegExp(task.title))).toBeDefined();
  });

  it('should invoke onClick', () => {
    const { getByText } = render(<Task task={task} onClick={onComplete} />);

    const component = getByText(/Learn React/);
    console.log('Component: ', component);

    // Click on Task title
    fireEvent.click(component);

    expect(onComplete).toHaveBeenCalledTimes(1);
  });
});
