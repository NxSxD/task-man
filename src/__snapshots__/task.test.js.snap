// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render correctly according to snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa jUIStS"
  >
    <span
      class="sc-bwzfXH iLItXQ"
    >
      Learn React
    </span>
    <span
      class="sc-htpNat hwgbR"
    >
      Learn react & redux
    </span>
  </div>
</div>
`;
